agent/coordinate_endpoint.go

```diff
@@ -85,22 +86,53 @@ func (s *HTTPServer) CoordinateNodes(resp http.ResponseWriter, req *http.Request
 		return nil, err
 	}
 
-	// Use empty list instead of nil.
-	if out.Coordinates == nil {
-		out.Coordinates = make(structs.Coordinates, 0)
+	return filterCoordinates(req, "", out.Coordinates), nil
+}
+
+// CoordinateNode returns the LAN node in the given datacenter, along with
+// raw network coordinates.
+func (s *HTTPServer) CoordinateNode(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	if req.Method != "GET" {
+		return nil, MethodNotAllowedError{req.Method, []string{"GET"}}
 	}
 
-	// Filter by segment if applicable
-	if v, ok := req.URL.Query()["segment"]; ok && len(v) > 0 {
-		segment := v[0]
-		filtered := make(structs.Coordinates, 0)
-		for _, coord := range out.Coordinates {
-			if coord.Segment == segment {
-				filtered = append(filtered, coord)
-			}
-		}
-		out.Coordinates = filtered
+	node := strings.TrimPrefix(req.URL.Path, "/v1/coordinate/node/")
+	args := structs.NodeSpecificRequest{Node: node}
+	if done := s.parse(resp, req, &args.Datacenter, &args.QueryOptions); done {
+		return nil, nil
+	}
+
+	var out structs.IndexedCoordinates
+	defer setMeta(resp, &out.QueryMeta)
+	if err := s.agent.RPC("Coordinate.Node", &args, &out); err != nil {
+		sort.Sort(&sorter{out.Coordinates})
+		return nil, err
+	}
+
+	return filterCoordinates(req, node, out.Coordinates), nil
+}
+
+func filterCoordinates(req *http.Request, node string, in structs.Coordinates) structs.Coordinates {
```

Comment:
Do we need to filter by node name? This seems like maybe a leftover from when this was built upon the existing endpoint?